### PR TITLE
Add EditScreen

### LINKED ISSUE #11: Create EditScreen component
This screen is responsible of editing/creating a specific item.

Create the small components using storybook (#7).
For the `EditScreen` component set a route using `react-router` named `#/edit`.

**Component:** `EditScreen`
This screen will show a `Header` (back, logo, save) and the _editable_ details of the item.
We will use 2 components to edit the item details: a `DatePicker` and `TextArea`.

props:
- `date`
- `text`
- `onBack`
- `onSave`

Notes: the elements used on the Header ("back", "save") will be an `ActionIcon` (#9).

**Component:** `DatePicker`
This component has a label ("When?"), a data and a picker that opens when you click/tap the date.

props:
- `date` <- the initial date to show when the component renders the first time
- `onChange` (as parameter of this function the picked date will be sent)

**Component:** `TextArea`

This component is composed of a normal html text area and a text length counter on the bottom that counts the current length of the details text area and the max length allowed (240).

props:
- `text` <- this is the initial text

exposed function:
- `getText`

### LINKED ISSUE #17: Create Modal component
**Component:** ```Modal```

This component will show a modal to expose a message to the user and eventually, request him/her to take a two-options decision (using ```Button```s - see #8).

The modal should be displayed in a layer over the screen with opacity to let the user see it but not interact with it (as any standard modal).

The modal should not dismiss when the user taps the screen, so it should has at least one ```Button``` to close it.

If a secondary ```Button``` is set, it should be displayed next to the primary ```Button```. For now, It will be placed on the left of the primary ```Button``` for all devices (Android like). This could be improved by displaying it on the right for iOS devices.

props:

- ```visible```: ```boolean```, deafult ```false```
- ```message```: ```string```, required
- ```primaryButton```: ```Button```, required
- ```secondaryButton```: ```Button```, default ```null```

### PR DESCRIPTION
**Pull request by fran-costa**
Close #11

### CODE REVIEW

**Comment by fran-costa** [comment]
Not ready for review.

**Comment by fran-costa** [comment]
Please, take a look at [this comment](https://github.com/Unixono/beautifulthings/issues/11#issuecomment-396632739).

**Comment by fran-costa** [comment]
In order to simplify the code review task, I splited the scope of this PR in two PRs, one (this one) to solve #11 and #17 to solve #12. As #11 needs #12 to be implemented, this PRs will be in standby until #17 gets merged.

**Comment summary:** Do not review this PR until #17 gets merged.

**Comment by fran-costa** [comment]
Not ready for review.

**Comment by fran-costa** [comment]
Ready for review

File: react-app/src/components/DatePicker/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      <div>
+        <label className={styles.label}>
+          When?
+        </label>
+        <input
+          type="date"
+          max={maxDate}
+          value={this.state.datePicked}
+          onChange={this._handleChange}
+        />
```
Comment (chiiph) [inline react-app/src/components/DatePicker/index.js]: I'm perfectly happy with this as the first iteration, but we should use an actual date picker here and not just an input eventually.

Comment (fran-costa) [reply react-app/src/components/DatePicker/index.js]: In mobile, at least in Android, is rendered as a DatePicker (the input acts as a button and opens a calendar to pick the date). **I don't know if this works in iOS...**

We could check this and see if we have to change it or not, what do you think?

Comment (chiiph) [reply react-app/src/components/DatePicker/index.js]: > We could check this and see if we have to change it or not, what do you think?

Nah, if it works on android, it's good enough for now.

Comment (fran-costa) [reply react-app/src/components/DatePicker/index.js]: Ok

**Pull request merged**